### PR TITLE
Minor simplifications to backend_svg.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backends/backend_svg.py [hunk 1]
```diff
@@ -1161,14 +1159,7 @@ def _draw_text_as_text(self, gc, x, y, s, prop, angle, ismath, mtext=None):
...
 
-                same_y = True
-                if len(chars) > 1:
-                    last_y = chars[0][1]
-                    for i in range(1, len(chars)):
-                        if chars[i][1] != last_y:
-                            same_y = False
-                            break
-                if same_y:
+                if len({y for x, y, t in chars}) == 1:
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py]: I think it's much clearer what this does...
I can do
```
same_y = (len(...) == 1)
if same_y: ...
```
I guess that gets the best of both worlds?

[the remark this answers is not in the document]
Comment (jklymak) [reply lib/matplotlib/backends/backend_svg.py]: Its the curly braces: I have no idea what that does.

[the remark this answers is not in the document]
Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py]: It's a set comprehension: https://docs.python.org/3/tutorial/datastructures.html#sets
The question is whether all y's are the same.  So you build a set with all y's and check whether its size is 1.

[the remark this answers is not in the document]
Comment (jklymak) [reply lib/matplotlib/backends/backend_svg.py]: OK, I see.  Each chars is 3-long, so x,y, t is juts unpacking the list.  Suggest short comment:

# check if all the y values (second element of each chars) are the same using set comprehension.

**Comment by anntzer** [comment]
added comment

File: lib/matplotlib/backends/backend_svg.py [hunk 2]
```diff
@@ -1005,9 +1004,8 @@ def _draw_text_as_path(self, gc, x, y, s, prop, angle, ismath, mtext=None):
 
             glyph_info, glyph_map_new, rects = _glyphs
 
-            # we store the character glyphs w/o flipping. Instead, the
-            # coordinate will be flipped when this characters are
-            # used.
+            # We store the character glyphs w/o flipping.  Instead, the
+            # coordinate will be flipped when this characters are used.
```
Comment (QuLogic) [inline lib/matplotlib/backends/backend_svg.py]: these characters or this character

**Comment by jklymak** [comment]
CI passed before minor typo in comment fix, so merging...

**Pull request merged**